search/utm_mgrs_coords_match.cpp

```diff
@@ -0,0 +1,216 @@
+#include "search/utm_mgrs_coords_match.hpp"
+
+#include "base/string_utils.hpp"
+
+#include "platform/utm_mgrs_utils.hpp"
+
+#include <iterator>
+#include <string>
+#include <utility>
+#include <math.h>
+
+#include "base/math.hpp"
+#include "base/string_utils.hpp"
+
+namespace search
+{
+
+using namespace std;
+
+string const kSpaceChars = " \t\r";
+
+size_t MatchZoneCode(string const & query, int &zone_code);
+size_t MatchZoneLetter(string const & query, char &zone_letter, size_t startPos);
+size_t MatchLong(string const & query, long &value, size_t startPos);
+
+// Parse UTM format "(\d\d)\s?(\W)\s+(\d+)\s+(\d+)" and converts it to lat,lon.
+// Return true if parsed successfully or false otherwise.
+// See utm_mgrs_coords_match_test.cpp for sample UTM strings
+bool MatchUTMCoords(string const & query, double & lat, double & lon)
+{
+  int zone_code;
+  char zone_letter;
+  long easting, northing;
+
+  size_t pos = MatchZoneCode(query, zone_code);
+  if (pos == string::npos)
+    return false;
+
+  pos = MatchZoneLetter(query, zone_letter, pos);
+  if (pos == string::npos)
+    return false;
+
+  pos = MatchLong(query, easting, pos);
+  if (pos == string::npos)
+    return false;
+
+  pos = MatchLong(query, northing, pos);
+  if (pos == string::npos)
+    return false;
+
+  return utm_mgrs_utils::UTMtoLatLon((double)easting, (double)northing, zone_code, zone_letter, lat, lon);
+}
+
+// Matches 2 digits zone code. Returns end position of matched chars or string::npos if no match.
+size_t MatchZoneCode(string const & query, int &zone_code)
```

Comment:
```suggestion
size_t MatchZoneCode(string const & query, int & zone_code)
```
There are many similar typos.

Reply:
Fixed formatting.